### ISSUE TITLE
Docker 4.0.1 Incompatible w/ Docker-Compose 1.24.0

### ISSUE THREAD

**Issue by mikeyamato** [Azure/iotedgehubdev#190]
I was following the [install instructions](https://github.com/Azure/iotedgehubdev#installing) for my Mac when I received the following message in my terminal after running step 4: 
```
docker-compose 1.24.0 has requirement docker[ssh]<4.0,>=3.7.0, but you'll have docker 4.0.1 which is incompatible.
```
Is this something I should be concerned about?

**Comment by adashen** [comment]
@mikeyamato Thanks for reporting the issue. The root cause is docker-compose has upgraded the version and the dependency has some conflict. We will fix the issue by updating the dependencies versions.

At mean time, to work around this issue. Can you try
1. Uninstall the docker "pip uninstall docker[ssh]"
2. Uninstall the iotedgehubdev "pip uninstall iotedgehubdev"
3. Install docker[ssh] to specific version "pip install docker[ssh]==3.7.0"
4. Install iotedgehubdev again "pip install -U iotedgehubdev"

**Comment by mikeyamato** [comment]
@adashen Please correct me if I implemented anything incorrectly but here's what I did: 
1. `pip uninstall docker`
2. `pip uninstall iotedgehubdev`
3. `pip install docker==3.7.0`
4. `pip install -U iotedgehubdev`

No errors received so I think I'm good? I've copied and pasted the output of each step below. 

**Step 1**
```
Uninstalling docker-4.0.1:
  Would remove:
    /usr/local/lib/python2.7/site-packages/docker-4.0.1.dist-info/*
    /usr/local/lib/python2.7/site-packages/docker/*
Proceed (y/n)? y
  Successfully uninstalled docker-4.0.1
```
**Step 2**
```
Uninstalling iotedgehubdev-0.8.0:
  Would remove:
    /usr/local/bin/iotedgehubdev
    /usr/local/lib/python2.7/site-packages/iotedgehubdev-0.8.0.dist-info/*
    /usr/local/lib/python2.7/site-packages/iotedgehubdev/*
Proceed (y/n)? y
  Successfully uninstalled iotedgehubdev-0.8.0
```
**Step 3**
```
Collecting docker==3.7.0
  Downloading https://files.pythonhosted.org/packages/7e/3c/b610f22b170b0f8fe4d8f78974878e116562389f666f99e6549567eb9d87/docker-3.7.0-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 9.1MB/s
Requirement already satisfied: six>=1.4.0 in /usr/local/lib/python2.7/site-packages (from docker==3.7.0) (1.12.0)
Requirement already satisfied: requests!=2.18.0,>=2.14.2 in /usr/local/lib/python2.7/site-packages (from docker==3.7.0) (2.20.1)
Requirement already satisfied: backports.ssl-match-hostname>=3.5; python_version < "3.5" in /usr/local/lib/python2.7/site-packages (from docker==3.7.0) (3.7.0.1)
Requirement already satisfied: ipaddress>=1.0.16; python_version < "3.3" in /usr/local/lib/python2.7/site-packages (from docker==3.7.0) (1.0.22)
Requirement already satisfied: websocket-client>=0.32.0 in /usr/local/lib/python2.7/site-packages (from docker==3.7.0) (0.56.0)
Collecting docker-pycreds>=0.4.0 (from docker==3.7.0)
  Downloading https://files.pythonhosted.org/packages/f5/e8/f6bd1eee09314e7e6dee49cbe2c5e22314ccdb38db16c9fc72d2fa80d054/docker_pycreds-0.4.0-py2.py3-none-any.whl
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/local/lib/python2.7/site-packages (from requests!=2.18.0,>=2.14.2->docker==3.7.0) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python2.7/site-packages (from requests!=2.18.0,>=2.14.2->docker==3.7.0) (2019.3.9)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/local/lib/python2.7/site-packages (from requests!=2.18.0,>=2.14.2->docker==3.7.0) (1.24.2)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/local/lib/python2.7/site-packages (from requests!=2.18.0,>=2.14.2->docker==3.7.0) (2.7)
Installing collected packages: docker-pycreds, docker
Successfully installed docker-3.7.0 docker-pycreds-0.4.0
```
**Step 4**
```
Collecting iotedgehubdev
  Using cached https://files.pythonhosted.org/packages/b2/ec/2692f21c2459490c01c7fb611426668ce91b380c229b9d0842bb58a6ff95/iotedgehubdev-0.8.0-py2.py3-none-any.whl
Requirement already satisfied, skipping upgrade: requests<2.21,>2.19.1 in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (2.20.1)
Requirement already satisfied, skipping upgrade: docker-compose>=1.21.0 in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (1.24.0)
Requirement already satisfied, skipping upgrade: pyOpenSSL>=17.0.0 in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (19.0.0)
Requirement already satisfied, skipping upgrade: pyyaml<4.3,>=3.10 in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (3.13)
Requirement already satisfied, skipping upgrade: click in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (7.0)
Requirement already satisfied, skipping upgrade: applicationinsights in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (0.11.9)
Requirement already satisfied, skipping upgrade: docker[ssh]>=3.6.0 in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (3.7.0)
Requirement already satisfied, skipping upgrade: six in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (1.12.0)
Requirement already satisfied, skipping upgrade: jsonpath-rw in /usr/local/lib/python2.7/site-packages (from iotedgehubdev) (1.4.0)
Requirement already satisfied, skipping upgrade: chardet<3.1.0,>=3.0.2 in /usr/local/lib/python2.7/site-packages (from requests<2.21,>2.19.1->iotedgehubdev) (3.0.4)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /usr/local/lib/python2.7/site-packages (from requests<2.21,>2.19.1->iotedgehubdev) (2019.3.9)
Requirement already satisfied, skipping upgrade: urllib3<1.25,>=1.21.1 in /usr/local/lib/python2.7/site-packages (from requests<2.21,>2.19.1->iotedgehubdev) (1.24.2)
Requirement already satisfied, skipping upgrade: idna<2.8,>=2.5 in /usr/local/lib/python2.7/site-packages (from requests<2.21,>2.19.1->iotedgehubdev) (2.7)
Requirement already satisfied, skipping upgrade: backports.ssl-match-hostname>=3.5; python_version < "3.5" in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (3.7.0.1)
Requirement already satisfied, skipping upgrade: ipaddress>=1.0.16; python_version < "3.3" in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (1.0.22)
Requirement already satisfied, skipping upgrade: websocket-client<1.0,>=0.32.0 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (0.56.0)
Requirement already satisfied, skipping upgrade: docopt<0.7,>=0.6.1 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (0.6.2)
Requirement already satisfied, skipping upgrade: jsonschema<3,>=2.5.1 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (2.6.0)
Requirement already satisfied, skipping upgrade: enum34<2,>=1.0.4; python_version < "3.4" in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (1.1.6)
Requirement already satisfied, skipping upgrade: cached-property<2,>=1.2.0 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (1.5.1)
Requirement already satisfied, skipping upgrade: texttable<0.10,>=0.9.0 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (0.9.1)
Requirement already satisfied, skipping upgrade: dockerpty<0.5,>=0.4.1 in /usr/local/lib/python2.7/site-packages (from docker-compose>=1.21.0->iotedgehubdev) (0.4.1)
Requirement already satisfied, skipping upgrade: cryptography>=2.3 in /usr/local/lib/python2.7/site-packages (from pyOpenSSL>=17.0.0->iotedgehubdev) (2.6.1)
Requirement already satisfied, skipping upgrade: docker-pycreds>=0.4.0 in /usr/local/lib/python2.7/site-packages (from docker[ssh]>=3.6.0->iotedgehubdev) (0.4.0)
Requirement already satisfied, skipping upgrade: paramiko>=2.4.2; extra == "ssh" in /usr/local/lib/python2.7/site-packages (from docker[ssh]>=3.6.0->iotedgehubdev) (2.4.2)
Requirement already satisfied, skipping upgrade: ply in /usr/local/lib/python2.7/site-packages (from jsonpath-rw->iotedgehubdev) (3.11)
Requirement already satisfied, skipping upgrade: decorator in /usr/local/lib/python2.7/site-packages (from jsonpath-rw->iotedgehubdev) (4.4.0)
Requirement already satisfied, skipping upgrade: functools32; python_version == "2.7" in /usr/local/lib/python2.7/site-packages (from jsonschema<3,>=2.5.1->docker-compose>=1.21.0->iotedgehubdev) (3.2.3.post2)
Requirement already satisfied, skipping upgrade: cffi!=1.11.3,>=1.8 in /usr/local/lib/python2.7/site-packages (from cryptography>=2.3->pyOpenSSL>=17.0.0->iotedgehubdev) (1.12.3)
Requirement already satisfied, skipping upgrade: asn1crypto>=0.21.0 in /usr/local/lib/python2.7/site-packages (from cryptography>=2.3->pyOpenSSL>=17.0.0->iotedgehubdev) (0.24.0)
Requirement already satisfied, skipping upgrade: pyasn1>=0.1.7 in /usr/local/lib/python2.7/site-packages (from paramiko>=2.4.2; extra == "ssh"->docker[ssh]>=3.6.0->iotedgehubdev) (0.4.5)
Requirement already satisfied, skipping upgrade: bcrypt>=3.1.3 in /usr/local/lib/python2.7/site-packages (from paramiko>=2.4.2; extra == "ssh"->docker[ssh]>=3.6.0->iotedgehubdev) (3.1.6)
Requirement already satisfied, skipping upgrade: pynacl>=1.0.1 in /usr/local/lib/python2.7/site-packages (from paramiko>=2.4.2; extra == "ssh"->docker[ssh]>=3.6.0->iotedgehubdev) (1.3.0)
Requirement already satisfied, skipping upgrade: pycparser in /usr/local/lib/python2.7/site-packages (from cffi!=1.11.3,>=1.8->cryptography>=2.3->pyOpenSSL>=17.0.0->iotedgehubdev) (2.19)
Installing collected packages: iotedgehubdev
Successfully installed iotedgehubdev-0.8.0
```

**Comment by adashen** [comment]
@mikeyamato Yes, iotedgehubdev now has been successfully installed

**Comment by SLdragon** [comment]
This issue has been fixed, update to 0.9.0 would solve the problem.
